snowflake/tests/test_unit.py

```diff
@@ -222,3 +222,27 @@ def test_emit_non_generic_tags_when_disabled(instance):
     tags = EXPECTED_TAGS + ['service_type:WAREHOUSE_METERING', 'service:COMPUTE_WH']
     normalised_tags = EXPECTED_TAGS + ['service_type:WAREHOUSE_METERING', 'snowflake_service:COMPUTE_WH']
     assert set(normalised_tags) == set(check._normalize_tags_type(tags))
+
+
+@pytest.mark.parametrize('aggregate_24_hours', [pytest.param(True), pytest.param(False)])
+def test_aggregate_24_hours_queries(aggregate_24_hours):
+    inst = {
+        'metric_groups': [
+            'snowflake.billing',
+            'snowflake.query',
+            'snowflake.data_transfer',
+            'snowflake.auto_recluster',
+            'snowflake.pipe',
+            'snowflake.replication',
+        ],
+        'user': 'user',
+        'password': 'password',
+        'account': 'account',
+        'role': 'role',
+    }
+    inst['aggregate_24_hours'] = aggregate_24_hours
+    check = SnowflakeCheck(CHECK_NAME, {}, [inst])
+
+    for query in check.metric_queries:
+        assert ('DATEADD(hour, -24, current_timestamp())' in query['query']) == aggregate_24_hours
```

Comment:
I'm not sure that it's the best way to do it since it would require changing the test every time the query is updated, but what do you thing asserting the whole query? The main concern I had about this change is that the `('DATEADD(hour, -24, current_timestamp())` could be inserted in the wrong place or whitespace could be off which messes up the query.

Reply:
There are also test failures for this test

Reply:
I've updated the test to make sure one query is correctly changed. The test will need update if this specific query is updated, but i don't think that's an issue as we don't update queries that often. Wdyt?

Reply:
Thanks, yeah that's true these will not be updated often.